client/app/bundles/course/assessment/pages/AssessmentStatistics/LiveFeedbackStatisticsTable.tsx

```diff
@@ -0,0 +1,306 @@
+
+  useEffect(() => {
+    const feedbackCounts = liveFeedbackStatistics
+      .flatMap((s) => s.liveFeedbackCount ?? [])
+      .map((c) => c ?? 0)
+      .filter((c) => c !== 0)
+      .sort((a, b) => a - b);
+    const upperQuartilePercentileIndex = Math.floor(
+      0.75 * (feedbackCounts.length - 1),
+    );
+    const upperQuartilePercentileValue =
+      feedbackCounts[upperQuartilePercentileIndex];
+    setUpperQuartileFeedbackCount(upperQuartilePercentileValue);
+
+    const filteredStats = includePhantom
+      ? liveFeedbackStatistics
+      : liveFeedbackStatistics.filter((s) => !s.courseUser.isPhantom);
+    setParsedStatistics(
+      filteredStats
+        .sort((a, b) => a.courseUser.name.localeCompare(b.courseUser.name))
+        .sort(
+          (a, b) =>
+            (b.liveFeedbackCount?.reduce(
+              (sum, count) => sum + (count || 0),
+              0,
+            ) ?? 0) -
+            (a.liveFeedbackCount?.reduce(
+              (sum, count) => sum + (count || 0),
+              0,
+            ) ?? 0),
+        )
+        .sort(
+          (a, b) =>
+            Number(a.courseUser.isPhantom) - Number(b.courseUser.isPhantom),
+        ),
+    );
+  }, [liveFeedbackStatistics, includePhantom]);
+
+  // the case where the live feedback count is null is handled separately inside the column
+  // (refer to the definition of statColumns below)
+  const renderNonNullAttemptCountCell = (count: number): ReactNode => {
+    const classname = getClassnameForLiveFeedbackCell(
+      count,
+      upperQuartileFeedbackCount,
+    );
+    return (
+      <div className={classname}>
+        <Box>{count}</Box>
+      </div>
+    );
+  };
+
+  // the customised sorting for grades to ensure null always is less than any non-null grade
+  const sortNullableLiveFeedbackCount = (
+    a: number | null,
+    b: number | null,
+  ): number => {
+    if (a === null && b === null) {
+      return 0;
+    }
+    if (a === null) {
+      return -1;
+    }
+    if (b === null) {
+      return 1;
+    }
+    return a - b;
+  };
+
+  const statColumns: ColumnTemplate<AssessmentLiveFeedbackStatistics>[] =
+    Array.from({ length: assessment?.questionCount ?? 0 }, (_, index) => {
+      return {
+        searchProps: {
+          getValue: (datum) =>
+            datum.liveFeedbackCount?.[index]?.toString() ?? '',
+        },
+        title: t(translations.questionIndex, { index: index + 1 }),
+        cell: (datum): ReactNode => {
+          return typeof datum.liveFeedbackCount?.[index] === 'number'
+            ? renderNonNullAttemptCountCell(datum.liveFeedbackCount?.[index])
+            : null;
+        },
+        sortable: true,
+        csvDownloadable: true,
+        className: 'text-right',
+        sortProps: {
+          sort: (a, b): number => {
+            return sortNullableLiveFeedbackCount(
+              a.liveFeedbackCount?.[index] ?? null,
+              b.liveFeedbackCount?.[index] ?? null,
+            );
```

Comment:
If the `sortNullableNumber` function is only used once throughout the codebase, might as well rewrite inline.

```ts
  const aValue = a.liveFeedbackCount?.[index] ?? Number.MIN_SAFE_INTEGER;
  const bValue = b.liveFeedbackCount?.[index] ?? Number.MIN_SAFE_INTEGER;
  
  return aValue - bValue;
```

Reply:
Done